packages/constellation/src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -0,0 +1,39 @@
+import '../styles.css'
+
+import React, { useEffect, useState } from 'react'
+
+const ThemeProvider = ({
+  children,
+  theme,
+  colorOverride = {},
+}: {
+  children: React.ReactNode
+  colorOverride: {
+    primary?: string
+  }
+  theme: string
+}) => {
+  const [colorOverrideStyles, setColorOverrideStyles] = useState({})
+
+  useEffect(() => {
+    let newColorOverrideStyles = {}
+
+    // eslint-disable-next-line array-callback-return
+    Object.keys(colorOverride).map((color) => {
+      newColorOverrideStyles = {
+        ...newColorOverrideStyles,
+        [`--color-${color}`]: colorOverride[color as keyof typeof colorOverride],
+      }
+    })
+
+    setColorOverrideStyles(newColorOverrideStyles)
+  }, [colorOverride])
+
+  return (
+    <div className={theme === 'dark' ? 'theme-dark' : 'theme-light'} style={colorOverrideStyles}>
```

Comment:
We may need to revisit this since we would be wrapping our entire app in a new div, it might break some of the children element styles.

Reply:
Yea definitely understand the concern here. @ethanjc I know you're in the process of documenting the approach to theming -- I think it's worth noting + addressing this concern.